lambdas/thumbnail/index.py

```diff
@@ -47,7 +54,146 @@
+    then rescales the intensities to fit on a int32 range.
+    """
+    # Set to float64 for futher correction math
+    img = img.astype(np.float64)
+
+    # Clip upper bound
+    img = np.clip(
+        img,
+        a_min=np.percentile(img, 0.01),
+        a_max=np.percentile(img, 99.99),
+    )
+
+    # Normalize greyscale values to floats between zero and one
+    img = img - np.min(img)
+    img = img / np.max(img)
+
+    # Cast the floats to integers
+    imax = np.iinfo(np.uint16).max + 1  # eg imax = 256 for uint8
+    img = img * imax
+    img[img == imax] = imax - 1
+    img = img.astype(np.int32)
+
+    return img
+
+
+def _format_n_dim_ndarray(img: AICSImage) -> np.ndarray:
+    # Even though the reader was n-dim, check if the actual data is simply greyscale and return
+    if len(img.reader.data.shape) == 2:
+        return img.reader.data
+
+    # Even the the reader was n-dim,
+    # check if the actual data is similar to YXC ("YX-RGBA" or "YX-RGB") and return
+    if (len(img.reader.data.shape) == 3 and (
+            img.reader.data.shape[2] == 3 or img.reader.data.shape[2] == 4)):
+        return img.reader.data
+
+    # Check which dimensions are available
+    # AICSImage makes strong assumptions about dimension ordering
+
+    # Reduce the array down to 2D + Channels when possible
+    # Always choose first Scene
+    if "S" in img.reader.dims:
+        img = AICSImage(img.data[0, :, :, :, :, :])
+    # Always choose middle time slice
+    if "T" in img.reader.dims:
+        img = AICSImage(img.data[0, img.data.shape[1] / 2, :, :, :, :])
+
+    # Keep Channel data, but max project when possible
+    if "C" in img.reader.dims:
+        projections = []
+        for i in range(img.data.shape[2]):
+            if "Z" in img.reader.dims:
+                # Add padding to the top and left of the projection
+                padded = np.pad(
+                    norm_img(img.data[0, 0, i, :, :, :].max(axis=0)),
+                    ((5, 0), (5, 0)),
+                    mode="constant"
+                )
+                projections.append(padded)
+            else:
+                # Add padding to the top and the left of the projection
+                padded = np.pad(
+                    norm_img(img.data[0, 0, i, 0, :, :]),
+                    ((5, 0), (5, 0)),
+                    mode="constant"
+                )
+                projections.append(padded)
+
+        # Get min grid shape
+        # For 6 channels this returns (2, 3)
+        min_grid_shape = choose_min_grid(len(projections))
+
+        # Make rows of images
+        rows = []
+        for y_i in range(min_grid_shape[0]):
+            row = []
+            for x_i in range(min_grid_shape[1]):
+                row.append(projections.pop(0))
+
+            rows.append(row)
+
+        # Concatinate each row then concatinate all rows together into a single 2D image
+        merged = []
+        for row in rows:
```

Comment:
This is mostly a matter of preference... But I'd either move this into the previous loop, or just do a list comprehension:
```
merged = [np.concatenate(...) for row in rows]
```